src/document-submit.js

```diff
@@ -23,12 +23,22 @@ import {
 } from './url';
 import {Services} from './services';
 import {dev, user} from './log';
+import {getElementServiceIfAvailableForDoc} from './element-service';
 
 /**
  * @param {!./service/ampdoc-impl.AmpDoc} ampdoc
+ * @return {!Promise}
  */
 export function installGlobalSubmitListenerForDoc(ampdoc) {
-  ampdoc.getRootNode().addEventListener('submit', onDocumentFormSubmit_, true);
+  // Register global submit event listener only if the amp-form
+  // extension is used. Allowing the usage of native forms, otherwise.
+  return getElementServiceIfAvailableForDoc(ampdoc, 'amp-form', 'amp-form')
```

Comment:
This will delay installation of the event listener until `amp-form` extension is installed. We should instead simply check if the extension *will be* installed and synchronously install or skip the event listener.

I think we can add a new function to element-service.js that uses `extensionScriptsInNode()` and rework `waitForExtensionIfPresent` to use it.

Reply:
done. Thanks